pkg/api/controller_test.go

```diff
@@ -4576,7 +4575,7 @@ func TestController_PrepareGarbageCollectionCommitted(t *testing.T) {
 }
 
 func TestController_ClientDisconnect(t *testing.T) {
-	handler, deps := setupHandlerWithWalkerFactory(t, store.NewFactory(nil))
+	handler, deps := setupHandler(t)
```

Comment:
Started from an attempt to add my tests here - 
Ended up not to, but on the way I did a small cleanup, which I think worth keeping (mainly in `server_test.go`.